pkg/lso/localvolumediscovery.go

```diff
@@ -0,0 +1,270 @@
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LocalVolumeDiscoveryBuilder provides a struct for localVolumeDiscovery object from the cluster
+// and a localVolumeDiscovery definition.
+type LocalVolumeDiscoveryBuilder struct {
+	// localVolumeDiscovery definition, used to create the localVolumeDiscovery object.
+	Definition *lsoV1alpha1.LocalVolumeDiscovery
+	// Created localVolumeDiscovery object.
+	Object *lsoV1alpha1.LocalVolumeDiscovery
+	// Used in functions that define or mutate localVolumeDiscovery definition. errorMsg is processed
+	// before the localVolumeDiscovery object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewLocalVolumeDiscoveryBuilder creates new instance of LocalVolumeDiscoveryBuilder.
+func NewLocalVolumeDiscoveryBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeDiscoveryBuilder {
+	glog.V(100).Infof("Initializing new localVolumeDiscovery structure with the following params: name: "+
+		"%s, namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// PullLocalVolumeDiscovery retrieves an existing localVolumeDiscovery object from the cluster.
+func PullLocalVolumeDiscovery(apiClient *clients.Settings, name, nsname string) (*LocalVolumeDiscoveryBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling localVolumeDiscovery object name:%s in namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("localVolumeDiscovery oject %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing localVolumeDiscovery from cluster.
+func (builder *LocalVolumeDiscoveryBuilder) Get() (*LocalVolumeDiscoveryBuilder, error) {
```

Comment:
This function should return *lsoV1alpha1.LocalVolumeDiscovery object instead of LocalVolumeDiscoveryBuilder